dbt-snowflake/src/dbt/adapters/snowflake/impl.py

```diff
@@ -83,6 +84,24 @@ class SnowflakeAdapter(SQLAdapter):
         }
     )
 
+    def __init__(self, config, mp_context) -> None:
+        super().__init__(config, mp_context)
+        self.add_catalog_integration(constants.DEFAULT_NATIVE_CATALOG)
+        self.add_catalog_integration(constants.DEFAULT_ICEBERG_CATALOG)
+
+    def add_catalog_integration(
+        self, catalog_integration: CatalogIntegrationConfig
+    ) -> CatalogIntegration:
+        # Snowflake uppercases everything in their metadata tables
+        # don't mutate the object that dbt-core passes in
+        catalog_integration = deepcopy(catalog_integration)
```

Comment:
Why is a deepcopy needed

Reply:
I don't want to mutate the integration itself, nor any of its attributes. If I understand normal copy correctly, the objects that the initial class points to are still the same objects on the copy, hence mutating those objects indirectly mutates the original object. For example, this works:
```python
from copy import copy
from dataclasses import dataclass


MY_CONFIG = {"name": "Fred"}


@dataclass
class MyObj:
    config: dict
    

my_first_obj = MyObj(config=MY_CONFIG)
my_second_obj = copy(my_first_obj)
my_second_obj.config["name"] = "Flintstone"
assert my_first_obj.config["name"] == "Flintstone"
```

Reply:
Yep your understanding is right. I was just curious about the extra resources taken for the deepcopy!